WordPress/gutenberg
# BoxControl: Passive deprecate `onMouseOver`/`onMouseOut`

Follow-up to https://github.com/WordPress/gutenberg/pull/67284#discussion_r1858633483

## What?

Passive deprecates (#61099) the `onMouseOver`/`onMouseOut` props on BoxControl. These props should be passed from the `inputProps` prop instead.

## Why?

This is an administrative change to streamline the props so they make more sense.

It is misleading and unnecessary for the `onMouseOver`/`onMouseOut` props to be exposed at the root level on BoxControl.  They can be passed through from the `inputProps` prop, and that is where they are ultimately passed anyway.

## Testing Instructions

1. In the `dimensions-panel.js` file, override the [`showSpacingPresetsControl` variable](https://github.com/miminari/gutenberg/blob/d57502ee6bad7bd6bf352a73700ba2e61d79c3ef/packages/block-editor/src/components/global-styles/dimensions-panel.js#L245) to `false`.
1. Insert a Paragraph block in the Block Editor.
1. In the block settings, activate the Margin control from the Dimensions section.
1. Hovering over the UnitControl should should a visual overlay of the margins on the block. This should be working like it is in `trunk`.

Technically I think the Padding control should also be working like this, but it doesn't quite work as expected for me in trunk either. Possibly related to #63191. It works once you commit a padding value by hitting the Enter key.
